### PR TITLE
feat: Process minidumps in subprocesses

### PR DESCRIPTION
**Pull request by mitsuhiko**
This moves minidump processing and demangling into subprocesses.

I would run this as an experiment first and see how the performance impact is. This does not yet use any pools because the async support does not yet allow that. Also demangling is incredibly inefficient because we spawn for each function name. This should become a batch job but that requires bigger changes in how the code is structured.

### CODE REVIEW

**Comment by jan-auer** [comment]
Requires https://github.com/mitsuhiko/procspawn/pull/27

File: src/actors/symbolication.rs [hunk 1]
```diff
@@ -1002,26 +1007,50 @@ impl SymbolicationActor {
...
+                            ))
+                        })
+                        .collect();
+
+                    Ok(procspawn::Json(cfi_modules))
+                },
+            );
 
-            Ok(cfi_modules)
+            // TODO(ja): This
```
Comment (tonyo) [inline src/actors/symbolication.rs:1042]: ?

File: .travis.yml [hunk 2]
```diff
@@ -14,6 +14,7 @@ branches:
     - master
     - /^release\/[\d.]+$/
     - /^deploy\/.*$/
+    - feature/subprocesses # Temporary
```
Comment (tonyo) [inline .travis.yml:17]: Let's remove this before merging.

Comment (flub) [inline .travis.yml:17]: Should be removed before merging

File: src/actors/symbolication.rs [hunk 3]
```diff
@@ -142,6 +146,7 @@ impl SymbolicationActor {
             cficaches,
             threadpool,
             requests,
+            spawnpool: Arc::new(procspawn::Pool::new(num_cpus::get()).expect("fixme")),
```
Comment (flub) [inline src/actors/symbolication.rs:149]: err, fixme.  maybe?

File: Cargo.toml [hunk 4]
```diff
@@ -15,7 +15,8 @@ failure = "0.1.6"
 serde = { version = "1.0.101", features = ["derive", "rc"] }
 futures = { version = "0.3", features = ["compat"] }
 futures01 = { version = "0.1.29", package = "futures" }
-bytes = "0.4.12"
+bytes = { version = "0.4.12", features = ["serde"] }
+procspawn = { features = ["backtrace", "json"], git = "https://github.com/flub/procspawn", branch = "features/sync-pool" }
```
Comment (jan-auer) [inline Cargo.toml:19]: Let's try to get this released before merging.

File: src/actors/symbolication.rs (hunk 3 above)
Comment (flub) [reply src/actors/symbolication.rs:149]: #218

**Pull request merged**